service/resource_router.go

```diff
@@ -0,0 +1,42 @@
+package service
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/aws-operator/service/key"
+)
+
+const (
+	legacyResourceName = "legacy"
+)
+
+// NewResourceRouter determines which resources are enabled based upon the
+// version in the version bundle.
+func NewResourceRouter(resources map[string][]framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+	return func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+		var enabledResources []framework.Resource
+
+		customObject, err := key.ToCustomObject(obj)
+		if err != nil {
+			return enabledResources, microerror.Mask(err)
+		}
+
+		switch key.VersionBundleVersion(customObject) {
+		case key.LegacyVersion:
+			// Legacy version so only enable the legacy resource.
+			enabledResources = resources[key.LegacyVersion]
```

Comment:
This is nice, but I am a bit 50/50 on the error handling. In theory I could make this panic now by being dumb. Some error handling would be good. Not sure if it is necessary. We should have tests for the routing though because this code will be maintained heavily over the lifetime of the operator. So it should become a habit to write tests when adding or removing version bundles. I can see executing the wrong version bundles being devastating. Then you probably also want to test what happens when you provide invalid versions where you then expect an error.

Reply:
Currently it returns the legacy resource if no version bundle is set which I want for compatibility with existing clusters.

I'll take another look at the error handling. I agree test coverage for this is important running the wrong version bundles will screw a cluster. I'll add some tests.

Reply:
Now returns an error if the version is unexpected.